plum/resolvable.py

```diff
@@ -41,6 +41,18 @@ def deliver(self, obj):
 
     def resolve(self):
         if self._obj is None:
-            raise ResolutionError("Promise was not kept.")
+            raise ResolutionError(f"Promise `{self!r}` was not kept.")
         else:
             return self._obj
+
+    @property
+    def resolved(self):
+        """bool: True if the Promise has been kept."""
+        return self._obj is not None
+
+    def __repr__(self):
+        # fallback in case superclasses don't overload repr
+        if self.resolved:
+            return f"Promise(obj={self._obj!r})"  # pragma: no cover
```

Comment:
Would you able to test the printing of `Promise` in e.g. a new test `tests/test_resolvable.py::test_promise_repr`?

Reply:
Moreover, should we change the constructor to

```python
    def __init__(self, obj=None):
        self._obj = obj
```

so that the output of `__repr__` can be `eval`d?

Reply:
Done